ES-1036: Ensure multicluster or unstable tests only run on the appropriate jobs

knguyen-r3 opened the pull request:
**Summary**
Add logic only trigger tests tagged with 'multicluster' when the runMultiClusterTests gradle property is passed similarly with "unstable" tag will only trigger unstable test. Else run all tests.

**Changes**
Updated build.gradle to allow granular test run when multicluster and unstable tag is use.

**Tested**
https://ci02.dev.r3.com/job/Corda5/job/corda-runtime-os-multi-cluster-tests/view/change-requests/job/PR-4393/
https://ci02.dev.r3.com/job/Corda5/job/corda-runtime-os-run-unstable/view/change-requests/job/PR-4393/

knguyen-r3 committed:
ES-1036: Enable Multi Cluster Tests Run

* Updated build.gradle to only run the multi-cluster tests.

knguyen-r3 committed:
Test

jennyang-r3 committed:
CORE-15818: upgrade runtime repo to Java 17 (#4372)

Update runtime repo to use JAVA 17 JDK, deprecating security manager due to the Java version upgrade.

kyriathar-r3 committed:
Revert "CORE-15806 Re-creating entity manager factories causing memory leak of non-heap (#4386)" (#4392)

Revert EMFs caching at reconciliations because it caused exhausting the DB connections

* This reverts commit 59e0cae6863b6083e9d6e6a3125980ca4df6915b

Tom-Fitzpatrick committed:
change plugins default Rest Client to use v5_1 (#4395)

Change plugins default Rest Client to use v5_1

ronanbrowne commented:
@knguyen-r3 can you rebase this branch with the latest from release/os/5.1 your PR is failing as it is missing this commit from the base branch which is now needed https://github.com/corda/corda-runtime-os/commits/release/os/5.1 due to a change in a lower layer of application stack (corda-cli)

Once the main PR you can see below is passing

trigger and link this job in description https://ci02.dev.r3.com/job/Corda5/job/corda-runtime-os-multi-cluster-tests/view/change-requests/job/PR-4393/ (you should see less tests ran on this one)

charlieR3 committed:
CORE-14784 Add e2e test config manager to update and revert configs per test. (#4363)

Added library to the e2e test utilities for testing with specific configurations. This provides common handling of configurations across e2e tests which includes reverting configuration. SingleClusterTestConfigManager can be use to manage the config of a specific or default single cluster, and MultiClusterTestConfigManager can be used to manage the same config across multiple clusters. Configuration overrides can be loaded for multiple properties and configuration sections before applying them to the cluster(s). The configuration manager implements AutoCloseable so these can be used with try-with-resources to ensure the configuration is reverted once no longer needed. Configuration will always be reverted back to it's original state before the test run.

A utility method is included called managedConfig which allows test writers to pass in a lambda which once complete will revert any test configuration set.

This also provides clear logging in the test output about what configuration values were set during a test run and confirmation that they were reverted. ClusterInfo was updated to expose a name property for clearer logging.

Two tests in this repo which were already testing with configuration changes have been updated to use the new library.

Commit:
Merge branch 'release/os/5.1' into knguyen/ES-1036/enable_multi_cluster_tests_run

ronanbrowne reviewed (dismissed):
LGTM

yarivus commented on build.gradle:
```diff
@@ -66,8 +66,13 @@ allprojects {
+                } else if (project.hasProperty('runUnstableTests')) {
+                    includeTag 'runUnstableTests'
+                } else {
+                  excludeTags 'runAllTestsNoExclusions'
```
```suggestion
                    excludeTags 'runAllTestsNoExclusions'
```

knguyen-r3 committed:
ES-1036: Ensure multicluster or unstable tests only run on the appropriate jobs

* Fix typo error for includeTags in build.gradle.

ronanbrowne reviewed (dismissed):
LGTM

knguyen-r3 committed:
Test

ronanbrowne reviewed (dismissed):
LGTM

ronanbrowne commented on build.gradle:
```diff
@@ -66,8 +66,13 @@ allprojects {
+                } else if (project.hasProperty('runUnstableTests')) {
+                    includeTags 'runUnstableTests'
+                } else {
+                    excludeTags 'MultiCluster', 'runUnstableTests'
```
```suggestion
                    excludeTags 'MultiCluster', 'Unstable'
```

we need to refrecn the unit tag here, which is `Unstable` , runUnstableTests is the gradle property which controls this

ronanbrowne replied:
I've applied this

Commit:
Update build.gradle

yarivus reviewed (commented):
thx @knguyen-r3
